src/components/annotations/draw.js

```diff
@@ -608,15 +663,22 @@ function drawRaw(gd, options, index, subplotId, xa, ya) {
                 },
                 moveFn: function(dx, dy) {
                     var csr = 'pointer';
+                    var xRefType = Axes.getRefType(options.xref);
+                    var yRefType = Axes.getRefType(options.yref);
```

Comment:
Could we possibly move this test to `shiftPosition` function?

Reply:
fixed by 2d25857bbe5ee314f7da07669ca4158f072bc831